lib/preview/routes/index.js

```diff
@@ -1,66 +1,130 @@
+          ${ domains ? getSiteList(domains, PORT) : '<p>No domains found</p>' }
+        </div>
+        <div style="width:100%;display:flex;flex-direction:row;justify-content:space-evenly">
+          <div>
+            <h1>Modules</h1>
+            ${ modulesHtml ? modulesHtml : `<p>No modules found in ${fakeDest}</p>` }
+          </div>
+          <div>
+            <h1>Templates</h1>
+            ${ templatesHtml ? templatesHtml : `<p>No templates found in ${fakeDest}</p>` }
+          </div>
+        </div>
+      </body>
+    </html>
   `;
 }
 
-const p = (text) => {
-  return `<p>${text}</p>`;
-}
-
-const getSiteList = (domains) => {
+const getSiteList = (domains, PORT) => {
   return listify(
     domains,
-    domainObj => `http://${domainObj.domain}.hslocal.net:3000`,
+    domainObj => `http://${domainObj.domain}.hslocal.net:${PORT}`,
     domainObj => domainObj.domain
   );
 }
 
-const getModuleList = (modules) => {
-  return listify(
-    modules.map(module => module.split('.')[0]),
-    moduleName => `http://hslocal.net:3000/module/${moduleName}`,
-    moduleName => moduleName
-  );
-}
-
-const getTemplateList = (templates) => {
-  return listify(
-    templates,
-    templateName => `http://hslocal.net:3000/template/${templateName}`,
-    templateName => templateName
-  )
-}
-
 const listify = (objects, hrefBuilder, labelBuilder) => {
   return '<ul>'
     + objects.reduce((x,y) => x += `<li><a href="${hrefBuilder(y)}">${labelBuilder(y)}</a></li>`, '')
     + '</ul>';
 }
 
+const getModulesForDisplayToUser = async (sessionInfo) => {
+  const { portalId, dest } = sessionInfo;
+  try {
+    const res = await fetchModulesPathStartsWith(portalId, dest);
+    const modulePaths = res
+      .objects
+      .map(moduleObj => moduleObj.path);
+    const filesGroupedByFolder = groupByFolder(modulePaths)
+    const htmlToRender = renderFilesGroupedByFolder(filesGroupedByFolder, 'module')
+    return htmlToRender;
+  } catch (err) {
+    console.log(`Failed to fetch modules for index page: ${err}`);
+    return undefined;
+  }
+}
+
+const getTemplatesForDisplayToUser = async (sessionInfo) => {
+  const { portalId, dest } = sessionInfo;
+  try {
+    const res = await fetchTemplatesPathStartsWith(portalId, dest);
+    const templatePaths = res
+      .objects
+      .filter(templateObj => templateObj.filename.split('.')[1] === 'html')
+      .map(templateObj => templateObj.path);
+    const filesGroupedByFolder = groupByFolder(templatePaths)
+    const htmlToRender = renderFilesGroupedByFolder(filesGroupedByFolder, 'template')
+    return htmlToRender;
+  } catch (err) {
+    console.log(`Failed to fetch templates for index page: ${err}`);
+    return undefined;
+  }
+}
+
+const groupByFolder = (paths) => {
+  return paths.reduce((acc, cur) => {
+    const { dir, base } = pathParse(cur);
+    if (Object.keys(acc).includes(dir)) {
+      acc[dir].push(base);
+    } else {
+      acc[dir] = [base];
+    }
+    return acc;
+  }, {})
+}
```

Comment:
Turns list of paths into paths grouped by folder, e.g 
```
[
"a/b/c/d.html",
"a/b/c/e.html",
"a/b/c/f/g.html",
]
```
becomes
```
{
"a/b/c": ["d.html", "e.html"],
"a/b/c/f": ["g.html"]
}
```
which makes the subsequent rendering to HTML a little easier